src/ResultSet.php

```diff
@@ -13,16 +17,21 @@ final class ResultSet implements \IteratorAggregate
      */
     private array $results = [];
 
-    public function addResult(string $attribute, Result $result): void
-    {
+    public function addResult(
+        string $attribute,
+        Result $result,
+        ?TranslatorInterface $translator = null,
```

Comment:
Usually all results within a set have the same translation language, domain and Translator.